packages/react-keytips/src/components/Keytips/Keytips.test.tsx

```diff
@@ -0,0 +1,192 @@
+import * as React from 'react';
+import { render, screen, act } from '@testing-library/react';
+import { Keytips } from './Keytips';
+import { useKeytipRef } from '../..';
+import { Button } from '@fluentui/react-components';
+import userEvent from '@testing-library/user-event';
+
+describe('Keytips', () => {
+  it('renders a default state', () => {
+    const rootId = 'ktp';
+    const content = 'Alt Windows';
+
+    const { baseElement } = render(<Keytips content={content} />);
+    const keytipsRoot = baseElement.querySelector(`#${rootId}`);
+    expect(keytipsRoot?.getAttribute('data-start-shortcut')).toEqual(content);
+  });
+
+  it('should have invisible and visible keytips', async () => {
+    const Component = () => {
+      const firstButton = useKeytipRef<HTMLButtonElement | HTMLAnchorElement>({
+        keySequences: ['b1'],
+        content: 'B1',
+      });
+
+      const secondButton = useKeytipRef<HTMLButtonElement | HTMLAnchorElement>({
+        keySequences: ['b2'],
+        content: 'B2',
+      });
+
+      return (
+        <>
+          <Keytips content="Alt Meta" />
+          <Button ref={firstButton}>1</Button>
+          <Button ref={secondButton}>2</Button>
+        </>
+      );
+    };
+
+    const { baseElement } = render(<Component />);
+
+    expect(screen.getByText('b1')).toBeTruthy();
+    expect(screen.getByText('b2')).toBeTruthy();
+    const portal = baseElement.querySelector('[data-portal-node]');
+
+    expect(portal?.children).toHaveLength(3);
+
+    expect(screen.queryAllByRole('tooltip')).toHaveLength(0);
+
+    // enter keytip mode
+    await act(async () => await userEvent.keyboard('{Alt>}{Meta}'));
+
+    expect(screen.getAllByRole('tooltip')).toHaveLength(2);
```

Comment:
this whole package is crucial to work correctly in real browsers (focus/keypresses etc) -> I'd suggest to add playwright component test and cover the real functionality there